exception_al/workshops/views.py

```diff
@@ -49,19 +53,41 @@ def get(self, request, pk):
 
     def put(self, request, pk):
         workshop = self.get_object(pk)
-        serializer = WorkshopSerializer(workshop, data=request.data)
+        serializer = WorkshopSerializer(data=request.data, context={'request': request})
+        # serializer = WorkshopSerializer(workshop, data=request.data)
         if serializer.is_valid():
             serializer.save()
             return Response(serializer.data)
         return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
     
+    def delete(self, request, pk):
+        workshop = self.get_object(pk)
+        workshop.delete()
+        return Response(status=status.HTTP_204_NO_CONTENT)
+    
+    
+
+class WorkshopDeleteView(APIView):
+    permission_classes = [permissions.IsAuthenticatedOrReadOnly]
```

Comment:
Please remove this view, as the WorkshopDetailView already contains Delete method.

Reply:
Done removed WorkshopDeleteView